subsys/fs/fat_fs.c

```diff
@@ -143,10 +158,23 @@ static ssize_t fatfs_read(struct fs_file_t *zfp, void *ptr, size_t size)
 
 static ssize_t fatfs_write(struct fs_file_t *zfp, const void *ptr, size_t size)
 {
-	FRESULT res;
+	FRESULT res = FR_OK;
 	unsigned int bw;
+	off_t pos = f_size((FIL *)zfp->filep);
+
+	/* FA_SEEKEND flag means that file has been opened for append.
+	 * The FAT FS write does not support the POSIX append semantics,
+	 * to always write at the end of file, so set file position
+	 * at the end before each write if FA_SEEKEND has been selected.
+	 */
+	if (((FIL *)zfp->filep)->flag & FA_SEEKEND) {
```

Comment:
Update to `FAX_APPEND` as noted above.

Reply:
Done as explained in https://github.com/zephyrproject-rtos/zephyr/pull/26305#discussion_r457908605